### PR TITLE
fix Demonstration tests

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)

Protobuf writing fails if the compressed observations are null. This fixes the exception during the tests.

### CODE REVIEW

**Comment by vincentpierre** [comment]
Since the user is supposed to write their own GtCompressedObservations, maybe we should check for null and raise?

File: com.unity.ml-agents/Runtime/Communicator/GrpcExtensions.cs [hunk 1]
```diff
@@ -229,11 +229,19 @@ public static ObservationProto GetObservationProto(this ISensor sensor, WriteAda
...
+                var compressedObs = sensor.GetCompressedObservation();
                 observationProto = new ObservationProto
                 {
-                    CompressedData = ByteString.CopyFrom(sensor.GetCompressedObservation()),
                     CompressionType = (CompressionTypeProto)sensor.GetCompressionType(),
                 };
+
+                // Passing a null byte[] to CompressedData will raise an exception, so leave it unset
+                // if the observations are null.
+                if (compressedObs != null)
```
Comment (surfnerd) [inline com.unity.ml-agents/Runtime/Communicator/GrpcExtensions.cs]: will null data be handled ok on the python side?

Comment (chriselion) [reply com.unity.ml-agents/Runtime/Communicator/GrpcExtensions.cs]: Probably not. I'll change this back to raise with a clearer error.

**Comment by chriselion** [comment]
Training and standalone tests are passing - going to merge this to get unit tests on master passing again.

**Pull request merged**